glean-core/android/src/main/java/mozilla/telemetry/glean/private/DatetimeMetricType.kt

```diff
@@ -114,11 +140,16 @@ class DatetimeMetricType(
      */
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testGetValueAsString(pingName: String = sendInPings.first()): String {
-        /*@Suppress("EXPERIMENTAL_API_USAGE")
+        @Suppress("EXPERIMENTAL_API_USAGE")
         Dispatchers.API.assertInTestingMode()
 
-        return DatetimesStorageEngine.getSnapshot(pingName, false)!![identifier]!!*/
-        return "LOL"
+        if (!testHasValue(pingName)) {
+            throw NullPointerException()
+        }
+        val ptr = LibGleanFFI
+            .INSTANCE
+            .glean_datetime_test_get_value_as_string(Glean.handle, this.handle, pingName)!!
+        return ptr.getAndConsumeRustString()
     }
 
     /**
```

Comment:
Note to self: the `testGetValue(pingName: String = sendInPings.first()): Date` must be implemented and will probably require importing `DateUtils.jt` from glean-ac. That's fine, as this testing API function is platform specific anyway.